api/v1alpha1/trafficmanagerendpoint_types.go

```diff
@@ -0,0 +1,164 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced,categories={fleet-networking},shortName=tme
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=`.spec.profile`,name="Profile",type=string
+// +kubebuilder:printcolumn:JSONPath=`.spec.endpointRef.name`,name="Endpoint-Reference",type=string
+// +kubebuilder:printcolumn:JSONPath=`.status.conditions[?(@.type=='Accepted')].status`,name="Is-Accepted",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+
+// TrafficManagerEndpoint is used to manage the Azure Traffic Manager Endpoints using cloud native way.
+// It could represent a group of endpoints in the multi-cluster scenario.
+// As the result, the controller may create multiple endpoints under the Traffic Manager Profile.
+// https://learn.microsoft.com/en-us/azure/traffic-manager/traffic-manager-endpoint-types
+type TrafficManagerEndpoint struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of TrafficManagerEndpoint.
+	Spec TrafficManagerEndpointSpec `json:"spec"`
+
+	// The observed status of TrafficManagerEndpoint.
+	// +optional
+	Status TrafficManagerEndpointStatus `json:"status,omitempty"`
+}
+
+type TrafficManagerEndpointSpec struct {
+	// +required
+	// immutable
+	// The profile should be created in the same namespace as the TrafficManagerEndpoint.
+	Profile *string `json:"profile"`
```

Comment:
do TrafficManagerEndpoint and Profile  have 1 :1 relation or N : 1?

Reply:
one profile can have multiple endpoints.